cosmo_tester/framework/util.py

```diff
@@ -414,44 +402,113 @@ def generate_ca_cert(ca_cert_path, ca_key_path):
     ])
 
 
-class ExecutionWaiting(Exception):
-    """
-    raised by `wait_for_execution` if it should be retried
-    """
+class ExecutionTimeout(Exception):
+    """Raised by `wait_for_execution` if the execution takes too long."""
```

Comment:
Just `An execution timed out` would be good enough of a docstring: it's common for exception descriptions to be just the description of the problem.

Here's some examples from the standard library:
```python
class IOError(EnvironmentError)
 |  I/O operation failed.

class AssertionError(StandardError)
 |  Assertion failed.

class KeyboardInterrupt(BaseException)
 |  Program interrupted by user.
```

Objects are not generally expected to know anything about their users, so `ExecutionTimeout` cannot really know where is it going to be thrown.

Yes, I'm aware this is just a reformat, so it's fiiiine if you don't change it, but, I see a chance to make a line shorter! woo

Reply:
Sure, let's make it prettier- pushed a commit for that.